src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -144,7 +144,7 @@ public bool ParseHeaders(TRequestHandler handler, ref SequenceReader<byte> reade
             var span = reader.UnreadSpan;
             while (span.Length > 0)
             {
-                var ch1 = (byte)0;
+                byte ch1;
                 var ch2 = (byte)0;
```

Comment:
why did it only change ch1 and not ch2?

Reply:
`ch2` isn't always set in the below code, `ch1` is because of the `TryRead` in the `else if`